runcoverage.sh

```diff
@@ -23,40 +23,67 @@ then
   fi
 fi
 
-$NUGET install -Verbosity quiet -OutputDirectory packages -Version 4.6.519 OpenCover
+$NUGET install -Verbosity quiet -OutputDirectory packages -Version 2017.1.20170613.162720 JetBrains.dotCover.CommandLineTools
 $NUGET install -Verbosity quiet -OutputDirectory packages -Version 2.4.5.0 ReportGenerator
-OPENCOVER=$PWD/packages/OpenCover.4.6.519/tools/OpenCover.Console.exe
+
+DOTCOVER=$PWD/packages/JetBrains.dotCover.CommandLineTools.2017.1.20170613.162720/tools/dotCover.exe
 REPORTGENERATOR=$PWD/packages/ReportGenerator.2.4.5.0/tools/ReportGenerator.exe
 FIND=/usr/bin/find
 
-rm -rf coverage
-mkdir coverage
+if [ ! -d coverage ]
+then
+ mkdir coverage
+fi
 
 while read csproj
 do
   testdir=$(dirname $csproj)
   api=$(basename $(realpath $testdir/..))
+
+  [ ! -d "$testdir/bin/Release/net452" ] && [ ! -d "$testdir/bin/Release/netcoreapp1.0" ] && continue
+
   echo "Running coverage for $api"
-  # OpenCover is picky about how it finds the excluded files. There may be a better approach than this,
-  # but it'll do for now.
-  generatedFiles=$($FIND $testdir/../$api -name '*.cs' | grep -v "obj/" | xargs -n 1 grep -l "// Generated" | sed 's/.*\/.*\//*\\\\*\\\\/g' | tr '\n' ';')
-  echo Coverage excluding files $generatedFiles
-  
-  $OPENCOVER \
-    -target:"c:\Program Files\dotnet\dotnet.exe" \
-    -targetargs:"test --no-build -f net452 -c Release $csproj" \
-    -mergeoutput \
-    -hideskipped:File \
-    -output:coverage/coverage.xml \
-    -filter:"+[$api]*" \
-    -oldStyle \
-    -searchdirs:$testdir/bin/$CONFIG/net452 \
-    -register:user \
-    -excludebyfile:$generatedFiles
 
+  echo "<CoverageParams>" > "coverage/$api.xml"
```

Comment:
i like it.
what do you think about including integration tests optionally as well?
We can make this script scan all coverage[Foo].xml files under the api and run each.

So when you run the generator, it will create just the version for tests.  But you are free to add additional coverage files to include other projects.